archivessnake/create_daos.py

```diff
@@ -0,0 +1,205 @@
+import pickle
+from configparser import ConfigParser
+from os.path import join
+
+import requests
+from asnake.aspace import ASpace
+
+# set up archivesspace client
+config = ConfigParser()
+config.read("local_settings.cfg")
+
+parser = argparse.ArgumentParser(
+    description="Using DIP information from Archivematica, adds DAOs to file-level children of a series in ArchivesSpace")
+parser.add_argument(
+    'resource', help='ArchivesSpace ID of resource. Example: 123')
+parser.add_argument('series', help='ArchivesSpace ID of series. Example: 4321')
+parser.add_argument(
+    'dip_info_file',
+    help='Path to pickle file containing array of dictionaries with DIP information.')
+args = parser.parse_args()
+
+
+def main(resource, series, dip_file):
+    """Main function, which is run when this script is executed"""
+    aspace = ASpace(
+        baseurl=config.get(
+            "ArchivesSpace", "baseURL"), username=config.get(
+            "ArchivesSpace", "username"), password=config.get(
+                "ArchivesSpace", "password"))
+    dao_url = join(config.get("ArchivesSpace", "baseURL"),
+                   "repositories/2/digital_objects")
+    dip_data = load_dip_file(dip_file)
+    for uri in get_uris(aspace, resource, series):
+        try:
+            dip = match_dip_to_aip(aspace, uri, dip_data)
+            dip_uuid, title, aip_uuid = get_dip_info(dip)
+            dao_uri = post_dao(dao_url, aspace, dip_uuid, title, aip_uuid)
+            update_component(aspace, uri, dao_uri)
+        except Exception as e:
+            print(e)
+
+
+def load_dip_file(dip_file):
+    """Uses pickle to open external file of DIP information, stored as an array of dictionaries.
+
+    Args:
+        dip_info (str): path to pickle file
+
+    Returns:
+        DIP information (array)
+    """
+    dip_data = pickle.load(open(dip_file, "rb"))
+    return dip_data
+
+
+def get_uris(aspace, resource, series):
+    """Takes a resource and a series from that resource to return all file-level descendants of the series that do not have linked digital objects.
+
+    Args:
+        aspace (asnake.aspace.ASpace): instantiation of ASpace
+        resource (str): ArchivesSpace ID of resource. Example: 123
+        series (str): ArchivesSpace ID of series. Example: 4321
+
+    Returns:
+        list of ArchivesSpace URIs (array)
+    """
+    lsrm_tree = aspace.client.get(
+        "repositories/2/resources/{}/tree".format(resource)).json()
+    lsrm_tree.get('children')[1].get('children')
+    series_tree = [child for child in lsrm_tree.get(
+        'children') if child.get("id") == series][0]
+    list_of_lsrm_uris = []
+    for subseries in series_tree.get("children"):
+        for file_level in subseries.get("children"):
+            if "digital_object" not in file_level.get("instance_types"):
+                list_of_lsrm_uris.append(file_level.get('record_uri'))
+    return list_of_lsrm_uris
+
+
+def match_dip_to_aip(aspace, ao_uri, dip_list):
```

Comment:
This function name is a little misleading - how about `match_dip_to_instance` or something like that?